packages/asc-ui/src/components/Radio/RadioStyle.ts

```diff
@@ -1,29 +1,142 @@
-import styled from '@datapunt/asc-core'
+import styled, { css } from '@datapunt/asc-core'
+import { themeColor, themeSpacing } from '../../utils'
+import { outlineStyle } from '../../utils/themeUtils'
 
-interface Props {
+enum Variants {
+  primary,
+  secondary,
+  tertiary,
+}
+
+export type Props = {
+  variant?: keyof typeof Variants
+  disabled?: boolean
+  error?: boolean
+  name?: string
+  value?: string
   id: string
-  name: string
-  value: string
-  defaultChecked: boolean
-  disabled: boolean
+  defaultChecked?: boolean
+  onChange?: Function
+}
+
+const getVariantColor = () => ({
+  variant,
+}: {
+  variant?: keyof typeof Variants
+}) => {
+  switch (variant) {
+    case 'primary':
+      return css`
+        ${themeColor('primary', 'main')};
+      `
+
+    case 'secondary':
+      return css`
+        ${themeColor('secondary', 'main')};
+      `
+
+    default:
+      return css`
+        ${themeColor('tint', 'level7')};
+      `
+  }
 }
 
-const RadioStyle = styled.input.attrs<Props>({
+const RadioStyle = styled.input.attrs({
   type: 'radio',
-  id: ({ id }: Props) => id,
-  name: ({ name }: Props) => name,
-  value: ({ value }: Props) => value,
-  defaultChecked: ({ defaultChecked }: Props) => defaultChecked,
-  disabled: ({ disabled }: Props) => disabled,
-})`
-  cursor: pointer;
-  height: 28px;
+})<Props>`
+  top: 0;
   left: 0;
+  width: 100%;
+  cursor: inherit;
+  height: 100%;
   margin: 0;
-  opacity: 0;
+  padding: 0;
   position: absolute;
-  top: 0;
-  width: 28px;
-  zoom: 1;
+  opacity: 0;
 `
+
 export default RadioStyle
+
+type StyleOnlyProps = {
+  variant?: keyof typeof Variants
+  checked?: boolean
+  disabled?: boolean
+  error?: boolean
+  focus?: boolean
+}
+
+const RadioCircleStyle = styled.span<StyleOnlyProps>`
+  width: 24px;
+  height: 24px;
+  color: ${themeColor('tint', 'level5')};
+  border: 1px solid;
+  border-radius: 50%;
+
+  &::after {
+    content: '';
+    position: absolute;
+    height: 16px;
+    width: 16px;
+    border-radius: 50%;
+    transform: translate(-50%, -50%);
+    left: 50%;
+    top: 50%;
+    background-color: ${getVariantColor()};
+    opacity: 0;
+    ${({ checked }) =>
+      checked &&
+      css`
+        opacity: 1;
+      `}
+  }
+
+  ${({ error, checked, disabled, focus }) =>
+    error &&
+    !checked &&
+    !disabled &&
+    !focus &&
+    css`
+      border-width: 2px;
+      color: red;
+    `}
+`
+
+export { RadioCircleStyle }
+
+const RadioWrapperStyle = styled.div<StyleOnlyProps & { focus: boolean }>`
+  position: relative;
+  display: inline-flex;
+  user-select: none;
+  vertical-align: middle;
+  flex-shrink: 0; /* IE11 fix */
+  padding: ${themeSpacing(1)};
+  margin-bottom: 1px;
```

Comment:
why the one pixel?

Reply:
![image](https://user-images.githubusercontent.com/7781865/69952751-22696980-14f8-11ea-9713-393e85909d54.png)
Because the spacing is 1px in design system (see screenshot)